### PR TITLE
Update 004-functions-classes.md

### PR DESCRIPTION
**Pull request by spamyak**
Changed explanation of getters and setters to improve accuracy as per MDN web docs. Small grammar and clarity improvements.

### CODE REVIEW

**Comment by MrBenJ** [comment]
Thanks! I haven't forgotten about this, just been heads down on other OSS projects to properly sit down and review this. Hang tight and I'll try to get it in soon.

File: book/004-functions-classes.md [hunk 1]
```diff
@@ -472,36 +472,34 @@ get isPaid() {
...
 
-Now let's say our client is awesome and paid us for our work (hooray!). Let's create a `setter` to set the `receivedOn` date.
+Let's create a `setter` for isPaid as well. We'll assume that if we set isPaid to true, we received payment today.
 
 ```js
 /**
   Pays an invoice and sets the date the invoice was paid.
-  @param  {Date}    date - The date the invoice was paid on
-  @return {Invoice}
+  @param  {paid}    paid - interpreted as a boolean
```
Comment (MrBenJ) [inline book/004-functions-classes.md]: This whole example might be nice, but it's starting to devalue what a setter can do and returning `this`. 

One thing I failed to mention is that by returning `this`, we're able to chain methods onto the object. I don't think this example needs to be changed, so if you can revert this change back to what it was, I think it would be better.

**Comment by MrBenJ** [comment]
@spamyak - Overall, great additions and catches, but I think that last example should go back to how it was. If you can just handle these little changes, I'd appreciate it a ton. 

Thanks!

**Comment by spamyak** [comment]
The problem is I don't think the original version of that last example works. Correct me if I'm doing something wrong here, but I tried the original example in the Chrome dev console:

```js
/**
 * class Invoice
 * Represents a single invoice in our backend
 *
 * @param {String} id          - Unique id of the Invoice
 * @param {String} payee       - Person/Company that owes us money
 * @param {Date}   dateCreated - Invoice creation date
 * @param {Date}   dueDate     - Date the amount is due
 * @param {Date}   receivedOn  - Date the amount was received
 * @param {Number} amount      - Amount due on invoice
 */
class Invoice {
  constructor(parameters) {
    const {
      id,
      payee,
      dateCreated,
      dueDate,
      receivedOn,
      amount
    } = parameters;

    this.id = id;
    this.payee = payee;
    this.dateCreated = dateCreated;
    this.dueDate = dueDate;
    this.receivedOn = receivedOn;
    this.amount = amount;
  }
  // Adding our new getter here
  get isPaid() {
    return Boolean(this.receivedOn);
  }
  /**
    Pays an invoice and sets the date the invoice was paid.
    @param  {Date}    date - The date the invoice was paid on
    @return {Invoice}
  */
  set pay(date) {
    this.paidOn = date;
    return this;
  }
}

let InvoiceA = new Invoice({id: 123123, payee: 'ACME Inc.', dateCreated: new Date('January 1, 2018, 07:00:00'), dueDate: new Date('January 1, 2019, 07:00:00'), amount: 5000.00})

// => undefined

InvoiceA.isPaid();

// => VM339:1 Uncaught TypeError: InvoiceA.isPaid is not a function    at <anonymous>:1:10 (anonymous) @ VM339:1

InvoiceA.isPaid;

// => false

InvoiceA.pay(new Date());

// => VM403:1 Uncaught TypeError: InvoiceA.pay is not a function    at <anonymous>:1:10 (anonymous) @ VM403:1

InvoiceA.pay

// => undefined

InvoiceA.isPaid

// => false

InvoiceA.pay = new Date();

// => Fri Nov 02 2018 20:30:11 GMT-0400 (Eastern Daylight Time)

InvoiceA.isPaid

// => false
```

I think the whole idea of Javascript getters and setters (at least using `get` and `set`) is that you use them to make pseudo-properties, so they don't work as functions accepting parameters. And my example is certainly not perfect, but it does work unlike the original.

```js
/**
 * class Invoice
 * Represents a single invoice in our backend
 *
 * @param {String} id          - Unique id of the Invoice
 * @param {String} payee       - Person/Company that owes us money
 * @param {Date}   dateCreated - Invoice creation date
 * @param {Date}   dueDate     - Date the amount is due
 * @param {Date}   receivedOn  - Date the amount was received
 * @param {Number} amount      - Amount due on invoice
 */
class Invoice {
  constructor(parameters) {
    const {
      id,
      payee,
      dateCreated,
      dueDate,
      receivedOn,
      amount
    } = parameters;

    this.id = id;
    this.payee = payee;
    this.dateCreated = dateCreated;
    this.dueDate = dueDate;
    this.receivedOn = receivedOn;
    this.amount = amount;
  }
  
  /**
    Returns a Boolean whether the invoice has been paid or not
    @return {Boolean} - true if paid, false if not paid
  */
  get isPaid() {
    return Boolean(this.receivedOn);
  }
  
  /**
    Pays an invoice and sets the date the invoice was paid.
    @param  {paid}    paid - interpreted as a boolean
  */
  set isPaid(paid) {
    if (paid) {
      this.receivedOn = new Date();
    } else {
      this.receivedOn = null;
    }
  }
}

let InvoiceA = new Invoice({id: 123123, payee: 'ACME Inc.', dateCreated: new Date('January 1, 2018, 07:00:00'), dueDate: new Date('January 1, 2019, 07:00:00'), amount: 5000.00});

// => undefined

InvoiceA.isPaid;

// => false

InvoiceA.isPaid = true;

// => true

InvoiceA.isPaid;

// => true

InvoiceA.receivedOn;

// => Fri Nov 02 2018 20:35:09 GMT-0400 (Eastern Daylight Time)
``

**Comment by MrBenJ** [comment]
I added in the fix for the class syntax myself, along with your previous edits @spamyak - thank you for your hard work. I really appreciate it :)
